serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java

```diff
@@ -584,7 +584,7 @@ private static String determineFieldType(ProtoType protoType, Schema schema) {
      * This method generates the synthetic one-of from a Proto3 optional field.
      */
     private static OneOf getProto3OptionalField(Field field) {
-        return new OneOf("_" + field.getName(), "", Collections.singletonList(field));
+        return new OneOf("_" + field.getName(), "", Collections.singletonList(field), field.getLocation(), field.getOptions());
```

Comment:
```suggestion
        return new OneOf("_" + field.getName(), "", Collections.singletonList(field), DEFAULT_LOCATION, new Options(ONEOF_OPTIONS, Collections.emptyList()));
```
(field options and oneof options are not the same)